cpp-client/build-dependencies.sh

```diff
@@ -191,6 +194,8 @@ while [ "$#" -ge 1 ]; do
     break
 done
 
+: ${GITHUB_BASE_URL:="https://github.com"}
```

Comment:
This is a naming nit. I believe once you abstract the name of the remote repository server, there's nothing "GitHub" about it, because Github is a brand name.  Maybe `REPO_SERVER_BASE_URL`

Reply:
We discussed over slack, since we have other code using the same variable changing it here now would have non-local impact.  We agreed to keep this as is for now.